user-docs/documentation/docs/how-to-guides/3-migration-guides/1-yarn-to-pnpm.md

```diff
@@ -0,0 +1,30 @@
+---
+sidebar_position: 1
+---
+
+# From Yarn to Pnpm
+
+[Yarn](https://yarnpkg.com/) and [Pnpm](https://pnpm.io/) are both package managers.
+
+## Why?
+
+In order to choose between these two alternatives, we have checked the following criteria:
+
+- built-in-support for monorepos: both package managers include this with workspaces
+- performance: pnpm reuses most of its package versions from its cache and symlinks them in our packages, which makes installs significantly faster
+- stability: by symlinking dependencies inside packages, pnpm better handles mismatches between package versions among our packages
+
+## How?
+
+Migrating from yarn to pnpm is quite straightforward:
```

Comment:
When I did the migration on my projet, I encountered some difficulties with missing `types` packages in dev dependencies.

For example: If two packages used `http-errors` as dependency and only one of them had the `@types/http-errors` as devDependency, the type checking was fine.

This lead to some issues with
- missing type packages
- conflicting type packages in some dependencies

This was the main non straightforward thing to sort out, but it may be worth it to mention it here ?

Reply:
Indeed, it may be a good idea. How did you sort it out?

Reply:
Two things
- Added the missing types packages for obvious ones
- Dig into dependencies to find out source issues in less obvious ones

Most often it was an issue with peer dependency which was not properly satisfied.

Reply:
I added a troubleshooting section in the doc